### PR TITLE
New Graph patterns

### PR DESCRIPTION
**Pull request by OlgaPodo**

### CODE REVIEW

File: graph/patterns/namespace.md [hunk 1]
```diff
@@ -0,0 +1,125 @@
+# Namespace
```
Comment (dkershaw10) [inline graph/patterns/namespace.md:1]: Or should we call this "Type namespaces"

File: graph/patterns/namespace.md [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+...
+</Schema>
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A public namespace must have "microsoft.graph.” prefix and be presented in camel
+case, i.e microsoft.graph.myNamespace.
+
+When type casting is required in the API query, request or response, a fully
+qualified type name is represented as concatenation of a namespace and a type
+name. For consistent user experience namespaces should be aligned with the corresponding API category path segment.
```
Comment (darrelmiller) [inline graph/patterns/namespace.md]:
```suggestion
name. For consistent user experience namespaces MUST be aligned with the corresponding API category path segment.
```

File: graph/patterns/namespace.md [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+
+Microsoft Graph API Design Pattern
+
+### *The Namespace provides the ability to organize resource definitions together into a logical set.*
+
+## Problem
+
+When building a complex offering API designers may need to model many different
+resources and their relationships. For better user experience and
+discoverability related API elements need to be grouped together.
```
Comment (dkershaw10) [inline graph/patterns/namespace.md:12]: elements or types?

File: graph/patterns/namespace.md [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+
+When type casting is required in the API query, request or response, a fully
+qualified type name is represented as concatenation of a namespace and a type
+name. For consistent user experience namespaces should be aligned with the corresponding API category path segment.
+
+
+## When to Use this Pattern
+
+API resource grouping creates a user-friendly experience keeping all resources
+for a specific feature close together. It also allows generating smallersmaller downloadable client components
```
Comment (PetRich-MSFT) [inline graph/patterns/namespace.md]:
```suggestion
for a specific feature close together. It also allows generating smaller downloadable client components 
```

Comment (darrelmiller) [inline graph/patterns/namespace.md]:
```suggestion
for a specific feature close together
```
This isn't true due to circular dependencies

File: graph/patterns/namespace.md [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+API designers can use the Namespace attribute of the CSDL schema to declare a
+namespace and logically organize related API entities in the Graph metadata.
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ XML
+<Schema Namespace="microsoft.graph.{namespace}">
+...
+</Schema>
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+A public namespace must have "microsoft.graph.” prefix and be presented in camel
```
Comment (dkershaw10) [inline graph/patterns/namespace.md]: Nit:  different quote marks :)

File: graph/patterns/namespace.md [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+introduced.
+
+## Issues and Considerations
+
+1.  Microsoft Graph consistency requirements discourage using the same type
+    names for different concepts even within different namespaces. Microsoft
+    Graph type names must be descriptive and unique within the API
+    surface without requiring  full qualification.
+
+2.  Namespace must be consistent with API category in the navigation path according to [Microsoft Graph REST API Guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/graph/GuidelinesGraph.md#uniform-resource-locators-urls)
```
Comment (PetRich-MSFT) [inline graph/patterns/namespace.md]: Should me have this a relative URL so the link won't break in a different branch?
```suggestion
2.  Namespace must be consistent with API category in the navigation path according to [Microsoft Graph REST API Guidelines](GuidelinesGraph.md#uniform-resource-locators-urls)
```

File: graph/patterns/namespace.md (hunk 4 above)
Comment (dkershaw10) [inline graph/patterns/namespace.md]: Nit: double smaller

File: graph/patterns/namespace.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+When type casting is required in the API query, request or response, a fully
+qualified type name is represented as concatenation of a namespace and a type
+name. For consistent user experience namespaces should be aligned with the corresponding API category path segment.
+
+
+## When to Use this Pattern
+
+API resource grouping creates a user-friendly experience keeping all resources
+for a specific feature close together. It also allows generating smallersmaller downloadable client components 
+and limits the length of IDE prompts such as Intellisense.
```
Comment (darrelmiller) [inline graph/patterns/namespace.md]:
```suggestion
and limits the length of IDE prompts such as auto-complete in some programming languages.
```

File: graph/patterns/namespace.md (hunk 4 above)
Comment (dkershaw10) [inline graph/patterns/namespace.md]: @darrelmiller should we be mentioning smaller downloadable client components?  Is that really true?

File: graph/patterns/namespace.md [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+
+We recommend creating a new namespace when a new top-level API category is
```
Comment (darrelmiller) [inline graph/patterns/namespace.md]: As discussed, I am not convinced this is what we should recommend.

File: graph/patterns/namespace.md [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+    names for different concepts even within different namespaces. Microsoft
+    Graph type names must be descriptive and unique within the API
+    surface without requiring  full qualification.
+
+2.  Namespace must be consistent with API category in the navigation path according to [Microsoft Graph REST API Guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/graph/GuidelinesGraph.md#uniform-resource-locators-urls)
+
+3.  When type name is ambiguous and requires a namespace qualifier, changing
+    namespace is a breaking change.
+
+4.  To extend a type in a different schema, a service must declare that schema
```
Comment (darrelmiller) [inline graph/patterns/namespace.md:57]: This seems like internal implementation details.

File: graph/patterns/namespace.md [hunk 10]
```diff
@@ -0,0 +1,125 @@
...
+## When to Use this Pattern
+
+API resource grouping creates a user-friendly experience keeping all resources
+for a specific feature close together. It also allows generating smallersmaller downloadable client components 
+and limits the length of IDE prompts such as Intellisense.
+
+We recommend creating a new namespace when a new top-level API category is
+introduced.
+
+## Issues and Considerations
```
Comment (dkershaw10) [inline graph/patterns/namespace.md:44]: You should call out that only one level of nesting deeper than microsoft.graph is supported, and that if you define something more deeply nested, the type will be coerced into the 3rd level namespace.

Is that implementation detail?

File: graph/patterns/namespace.md [hunk 11]
```diff
@@ -0,0 +1,125 @@
...
+    namespace is a breaking change.
+
+4.  To extend a type in a different schema, a service must declare that schema
+    and the type in it. This is conceptually similar to .NET partial types.
+
+5.  To reference a type in a different schema, simply refer to that type by
+    fully qualified name (namespace + type name).
+
+6.  Cyclical references between namespaces are not allowed as many
+    object-oriented languages don’t support a cycle in inheritance.
```
Comment (darrelmiller) [inline graph/patterns/namespace.md]:
```suggestion
    object-oriented languages don’t support a cycles between namespaces.
```
The constraint is not related to specifically to inheritance

File: graph/patterns/namespace.md [hunk 12]
```diff
@@ -0,0 +1,125 @@
...
+…
+</Schema>
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Fully qualified type name: microsoft.graph.search.bookmark
+
+### Managing multiple schemas:
+
+Workloads must define schemas in their csdl using the Edmx format.
+[Microsoft.IC3.DataPlatform](https://dev.azure.com/msazure/One/_git/AD-AggregatorService-Workloads?path=%2FWorkloads%2FMicrosoft.IC3.DataPlatform%2Foverride%2Fschema-Prod-beta.csdl)
```
Comment (dkershaw10) [inline graph/patterns/namespace.md]: This doesn't seem like this should be in the public docs.

File: graph/patterns/namespace.md (hunk 3 above)
Comment (OlgaPodo) [reply graph/patterns/namespace.md:12]: elements will include types and operations

File: graph/patterns/namespace.md (hunk 9 above)
Comment (OlgaPodo) [reply graph/patterns/namespace.md:57]: it doesn't look like AGS specific consideration and is used quite often for graph.entity. <EntityType Name="callRecord" BaseType="graph.entity"

File: graph/patterns/namespace.md (hunk 8 above)
Comment (OlgaPodo) [reply graph/patterns/namespace.md]: changed to aligned as discussed at API Council

File: graph/patterns/namespace.md (hunk 10 above)
Comment (OlgaPodo) [reply graph/patterns/namespace.md:44]: added as a constraint for clarity

**Pull request merged**